### DISCUSSION TITLE
Incremental Deployment

### DISCUSSION THREAD

**Discussion by SomanathSankaran** [Azure-Player/azure.datafactory.tools#190] [Q&A] [answered]
Hi Team,

we are planning using adf task for CD Deployment.
We need couple of clarification reg how we can do incremental Deployment for below scenarios
1.Directly deploy only selected pipelines automatically based on commit once in 2 weeks rather than manually including files
Does this tool have a mechanism to automatically deploy changed files alone or every time we have to do it manually
2.Is it possible to have idempotent deployment like ARM Template or every time if we deploy as a whole (if we dont have any way for step 1 ) 
3.Is it possible to detect datasets,LS etc automatically which are not being used(like one used for test) rather than manually adding the files every time

**Comment by NowinskiK** [comment, marked as answer]
Hi @SomanathSankaran 
Incremental deployment is not available out of the box.
1. It has nothing to do with a repository where you keep the code of your ADF, so there is no method to determine files changed for the last 2 weeks.
2. #adftools gives you an option of selective deployment - take it or leave it. It's up to you in which scenario you might want to use it. If you deploy entire ADF it will work as ARM Template.
3. What do you mean by "not being used"? Orphaned objects? If so, such a method is not yet available in the tool.
